src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1109,14 +1115,20 @@ private TokenValidationResult ValidateTokenPayload(JsonWebToken jsonWebToken, To
             }
             Validators.ValidateIssuerSecurityKey(jsonWebToken.SigningKey, jsonWebToken, validationParameters);
             var type = Validators.ValidateTokenType(jsonWebToken.Typ, jsonWebToken, validationParameters);
-
-            return new TokenValidationResult
+            var claimsIdentity = CreateClaimsIdentity(jsonWebToken, validationParameters, issuer);
+            var tokenValidationResult = new TokenValidationResult
             {
                 SecurityToken = jsonWebToken,
-                ClaimsIdentity = CreateClaimsIdentity(jsonWebToken, validationParameters, issuer),
+                ClaimsIdentity = claimsIdentity,
                 IsValid = true,
                 TokenType = type
             };
+
+            var claims = JwtTokenUtilities.CreateDictionaryFromClaims(claimsIdentity.Claims);
+            foreach (var claim in claims)
```

Comment:
@RojaEnnam  We should delay performing this until it is asked for as this will have a performance impact on all call graphs.

Reply:
@brentschmaltz are you saying we should add the property but not set it on TokenValidationResult ?